### PR TITLE
Reductions updates to fix naming and arg ordering, add unsigned types

### PR DESCRIPTION
**Pull request by gmegan**
Addressing changes to reductions as per [Teams WG 2019-3-14](Teams-WG-2019-3-14)

### CODE REVIEW

**Comment by nspark** [comment]
I noticed the other day that the existing SHMEM reductions use the signed integer type `int` for the number of reduction elements `nreduce`. This seemed odd to me, given that SHMEM takes `size_t`-typed counts for every other comparable argument (e.g., `nelems` or `nbytes`).

Does anyone know of any historical basis for this choice of type? (@bcernohous, maybe?) Do we want to switch to `size_t` for the new team-based reductions?

File: content/shmem_reductions.tex [hunk 1]
```diff
@@ -101,7 +106,7 @@
 %% C11
 {\color{Green}
 \begin{C11synopsis}
-int @\FuncDecl{shmem\_xor\_to\_all}@(TYPE *dest, const TYPE *source, int nreduce, shmem_team_t team);
+int @\FuncDecl{shmem\_xor\_reduce}@(shmem_team_t team, TYPE *dest, const TYPE *source, int nreduce;
```
Comment (nspark) [inline content/shmem_reductions.tex]: Missing a closing right-parenthesis here

Comment (gmegan) [reply content/shmem_reductions.tex]: Fixed. Thank you.

**Comment by gmegan** [comment]
> I noticed the other day that the existing SHMEM reductions use the signed integer type `int` for the number of reduction elements `nreduce`. This seemed odd to me, given that SHMEM takes `size_t`-typed counts for every other comparable argument (e.g., `nelems` or `nbytes`).

I agree, this does seem odd. Looking back at the original 1.0 spec, it has always been int and nelems has always been size_t. Maybe this has something to do with Fortran type compatibility and/or indexing into the pWrk array? I will ask the list about this to see if there is some reason to leave it as int. Otherwise size_t seems better.

**Pull request merged**